### PR TITLE
Add MISRA config file

### PR DESCRIPTION
**Pull request by aggarw13**
Add a MISRA config file to repository for CI Coverity checks﻿

### CODE REVIEW

**Review by muneebahmed10 — APPROVED** [review]
Approving since it's a copy of the file in the hub repo: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/tools/coverity/misra.config

File: tools/coverity/misra.config [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+{
+    version : "2.0",
+    standard : "c2012",
+    title: "Coverity MISRA Configuration",
+    deviations : [
+        // Disable the following rules.
+        {
+            deviation: "Directive 4.5",
+            reason: "Allow names that MISRA considers ambiguous (such as enum IOT_MQTT_CONNECT and function IotMqtt_Connect)."
```
Comment (sarenameas) [inline tools/coverity/misra.config:11]: Amit asked Daniel to change this example here

File: tools/coverity/misra.config [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            deviation: "Rule 3.1",
+            reason: "Allow nested comments. Documentation blocks contain comments for example code."
+        },
+        {
+            deviation: "Rule 11.5",
+            reason: "Allow casts from void *. Contexts are passed as void * and must be cast to the correct data type before use."
+        },
+        {
+            deviation: "Rule 21.1",
```
Comment (sarenameas) [inline tools/coverity/misra.config:38]: Update everything as mentioned here: https://github.com/FreeRTOS/coreMQTT/blob/master/MISRA.md

**Comment by muneebahmed10** [comment]
With the addition of this file, should we change [this link](https://github.com/FreeRTOS/coreMQTT/blob/master/MISRA.md#ignored-by-coverity-configuration) in MISRA.md to be a relative link to this?

**Review by muneebahmed10 — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request closed without merging**